### PR TITLE
Added verifySignature function in PublicAccount class

### PR DESCRIPTION
**Pull request by aizaiz**

### CODE REVIEW

File: src/model/account/PublicAccount.ts [hunk 1]
```diff
@@ -53,6 +54,48 @@ export class PublicAccount {
...
+    /**
+     * Verify a signature.
+     *
+     * @param {string} publicKey - The public key to use for verification.
+     * @param {string} data - The data to verify.
+     * @param {string} signature - The signature to verify.
+     *
+     * @return {boolean}  - True if the signature is valid, false otherwise.
+     */
+    static verifySignature(publicKey: string, data: string, signature: string): boolean {
```
Comment (aleixmorgadas) [inline src/model/account/PublicAccount.ts]: changing `publicKey: string` by `publicAccount: PublicAccount` will remove the next code,

```javascript
if (publicKey.length !== 64 && publicKey.length !== 66) {
    throw new Error('Not a valid public key');
}
```
making the design safer.

Comment (aleixmorgadas) [reply src/model/account/PublicAccount.ts]: `signature` has some constrains too, I'm thinking in create a class to represent the signature and add there the invariants. It should remove the:

```javascript
if (signature.length !== 128) {
    throw new Error('Signature length is incorrect !');
}
// ...
const _signature = convert.hexToUint8(signature);
```
Also, it hasn't been checked that the signature is an hexadecimal string.

**Comment by aizaiz** [comment]
halo @Vektrat, thank you for the input. regarding the test, I will consider it later after I finish this first, because I have another important stuff to do.

@aleixmorgadas so what is the next step? should I add some change base on @Vektrat and your input and create another pull request?

**Comment by aleixmorgadas** [comment]
> @aleixmorgadas so what is the next step? should I add some change base on @Vektrat and your input and create another pull request?

Take the input from @Vektrat, and commit to your branch, this PR will be updated automatically, so we will be able to see the latest changes.

File: src/model/account/PublicAccount.ts (hunk 1 above)
Comment (aizaiz) [reply src/model/account/PublicAccount.ts]: > Also, it hasn't been checked that the signature is an hexadecimal string

well, this "if condition" below made sure that the signature must be hexadecimal

```
if (convert.isHexString(signature)) {
        throw new Error('Signature must be hexadecimal only');
}
```

**Comment by aizaiz** [comment]
@aleixmorgadas I have pushed a new commit, please review it. I changed publicKey to publicAccount but it's a little bit weird because the function it self is inside PublicAccount class. I think we should move the function to an utility class.

**Comment by aleixmorgadas** [comment]
The next step should be adding the tests. They should guide the usage and then we find out if it has to belong to another class or not.

**Comment by aizaiz** [comment]
@aleixmorgadas I added some test
but I got errors from nem2-library code

```
4 failing

  1) Signature verification
       Can verify a signature:
     TypeError: array.every is not a function
      at Object.isZero (node_modules/nem2-library/dist/coders/array.js:61:16)
      at Object.verify (node_modules/nem2-library/dist/crypto/keyPair.js:162:24)
      at Object.verify (node_modules/nem2-library/dist/crypto/keyPair.js:256:25)
      at Function.verifySignature (dist/src/model/account/PublicAccount.js:82:39)
      at Context.it (dist/test/model/account/PublicAccount.spec.js:36:53)

  2) Signature verification
       Return false if wrong public key provided:
     TypeError: array.every is not a function
      at Object.isZero (node_modules/nem2-library/dist/coders/array.js:61:16)
      at Object.verify (node_modules/nem2-library/dist/crypto/keyPair.js:162:24)
      at Object.verify (node_modules/nem2-library/dist/crypto/keyPair.js:256:25)
      at Function.verifySignature (dist/src/model/account/PublicAccount.js:82:39)
      at Context.it (dist/test/model/account/PublicAccount.spec.js:61:53)

  3) Signature verification
       Return false if data is not corresponding to signature provided:
     TypeError: array.every is not a function
      at Object.isZero (node_modules/nem2-library/dist/coders/array.js:61:16)
      at Object.verify (node_modules/nem2-library/dist/crypto/keyPair.js:162:24)
      at Object.verify (node_modules/nem2-library/dist/crypto/keyPair.js:256:25)
      at Function.verifySignature (dist/src/model/account/PublicAccount.js:82:39)
      at Context.it (dist/test/model/account/PublicAccount.spec.js:69:53)

  4) Signature verification
       Return false if signature is not corresponding to data provided:
     TypeError: array.every is not a function
      at Object.isZero (node_modules/nem2-library/dist/coders/array.js:61:16)
      at Object.verify (node_modules/nem2-library/dist/crypto/keyPair.js:162:24)
      at Object.verify (node_modules/nem2-library/dist/crypto/keyPair.js:256:25)
      at Function.verifySignature (dist/src/model/account/PublicAccount.js:82:39)
      at Context.it (dist/test/model/account/PublicAccount.spec.js:77:53)

```

**Pull request merged**